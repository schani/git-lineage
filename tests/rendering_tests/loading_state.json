{
  "active_panel": "Inspector",
  "selected_file_navigator_index": null,
  "search_query": "",
  "in_search_mode": false,
  "commit_list": [
    {
      "hash": "def456ghi789def456ghi789def456ghi789def4",
      "short_hash": "def456g",
      "author": "JS Dev",
      "date": "30 mins ago",
      "subject": "Add async functions"
    }
  ],
  "selected_commit_index": 0,
  "current_content": [
    "async function fetchData() {",
    "  const response = await fetch('/api/data');",
    "  return response.json();",
    "}",
    "",
    "function processData(data) {",
    "  return data.map(item => item.value);",
    "}"
  ],
  "cursor_line": 1,
  "cursor_column": 20,
  "inspector_scroll_vertical": 0,
  "inspector_scroll_horizontal": 0,
  "show_diff_view": false,
  "status_message": "Searching for next change...",
  "is_loading": true,
  "file_tree": {
    "root": [
      {
        "name": "large_file.js",
        "path": "./large_file.js",
        "is_dir": false,
        "git_status": "M",
        "is_expanded": false,
        "children": [],
        "parent_path": null
      }
    ],
    "repo_root": "",
    "git_status_map": {}
  },
  "active_file_context": "./large_file.js",
  "selected_commit_hash": "def456ghi789def456ghi789def456ghi789def4"
}