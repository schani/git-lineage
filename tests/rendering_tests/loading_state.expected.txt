┌ File Navigator ──────────┐┌ Code Inspector - large_file.js @ def456gh ───────┐
│M large_file.js           ││   1 async function fetchData() {                 │
│                          ││   2   const response = await fetch('/api/data'); │
│                          ││   3   return response.json();                    │
│                          ││   4 }                                            │
│                          ││   5                                              │
│                          ││   6 function processData(data) {                 │
│                          ││   7   return data.map(item => item.value);       │
│                          ││   8 }                                            │
│                          ││                                                  │
│                          ││                                                  │
│                          ││                                                  │
└──────────────────────────┘│                                                  │
┌ Commit History (large_fil┐│                                                  │
│>> def456g 30 mins ago JS ││                                                  │
│   More commits available ││                                                  │
│                          ││                                                  │
│                          ││                                                  │
│                          ││                                                  │
│                          ││                                                  │
│                          ││                                                  │
│                          ││                                                  │
│                          ││                                                  │
│                          ││                                                  │
Loading... | Searching for next change... | Tab: Switch panel | 1/2/3: Direct pa
